kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java

```diff
@@ -1,37 +1,49 @@
 package uk.gov.homeoffice.digital.sas.kafka.listener;
 
-import jakarta.persistence.PostPersist;
-import jakarta.persistence.PostRemove;
-import jakarta.persistence.PostUpdate;
+import jakarta.persistence.PrePersist;
+import jakarta.persistence.PreRemove;
+import jakarta.persistence.PreUpdate;
 import jakarta.validation.constraints.NotNull;
+import java.util.function.BiConsumer;
+import org.springframework.beans.factory.annotation.Autowired;
 import uk.gov.homeoffice.digital.sas.kafka.message.KafkaAction;
 import uk.gov.homeoffice.digital.sas.kafka.message.Messageable;
 import uk.gov.homeoffice.digital.sas.kafka.producer.KafkaProducerService;
+import uk.gov.homeoffice.digital.sas.kafka.transactionsync.KafkaDbTransactionSynchronizer;
 
 public final class KafkaEntityListener<T extends Messageable> {
 
   private final KafkaProducerService<T> kafkaProducerService;
 
+  @Autowired
+  private KafkaDbTransactionSynchronizer kafkaDbTransactionSynchronizer;
```

Comment:
I see that the Timecard version didn't have a constructor here so this dependency was field injected, maybe we should autowire in the constructor along with the producerService. (Since we already have a constructor here)

Reply:
Actioned

Reply:
could we also make it final please

Reply:
Yes I have done this now.